File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        if (k > nums.size()) {
            k = k % nums.size();
        }

        vector<int> ans;
        for (int i = nums.size() - k; i < nums.size(); i++) {
            ans.push_back(nums[i]);
        }
        for (int i = 0; i < nums.size() - k; i++) {
            ans.push_back(nums[i]);
        }
        nums = ans;
    }
};

File: 0189-rotate-array/0189-rotate-array.java
class Solution {
        
    public void reverse(int[] nums, int st, int end){
        int tmp = 0;
            while(st<end){
                tmp = nums[st];
                nums[st] = nums[end];
                nums[end] = tmp;
                st++;
                end--;    
            }
    }    
        
    public void rotate(int[] nums, int k) {
         k = k % nums.length;
         reverse(nums, 0, nums.length - k - 1);
         reverse(nums, nums.length - k, nums.length - 1);
         reverse(nums, 0, nums.length - 1);   
    }
}